behaviors/magic-button.js

```diff
@@ -0,0 +1,168 @@
+const _ = require('lodash'),
+  site = require('../services/site'),
+  dom = require('../services/dom'),
+  getInput = require('../services/get-input'),
+  transformers = {
+    // this is an object of available transforms
+    // components can specify which transform they want to use in their schemae
+    mediaplayUrl: function (data) {
+      const path = data.replace(/^.*?imgs\//, ''); // remove domain and everything up to imgs/
+
+      // remove rendition stuff
+      let barePath = path.replace(/\.w\d+/, ''); // remove width
+
+      barePath = barePath.replace(/\.h\d+/, ''); // remove height
+      barePath = barePath.replace('.2x', ''); // remove resolution
+
+      return barePath;
+    }
+  };
```

Comment:
Wondering how this will grow. Should the transformer be passed into the behavior?